File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/Conne/ProductColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ProductRow } from "./ProductRow";

export interface ProductColumns {
    Id: Column<ProductRow>;
    Name: Column<ProductRow>;
    Description: Column<ProductRow>;
    Price: Column<ProductRow>;
    CatidName: Column<ProductRow>;
    Photo: Column<ProductRow>;
}

export class ProductColumns extends ColumnsBase<ProductRow> {
    static readonly columnsKey = 'Conne.Product';
    static readonly Fields = fieldsProxy<ProductColumns>();
}

File: AdminPanel/AdminPanel.Web/Modules/Conne/Product/ProductColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel.Conne.Columns;

[ColumnsScript("Conne.Product")]
[BasedOnRow(typeof(ProductRow), CheckNames = true)]
public class ProductColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Description { get; set; }
    public decimal Price { get; set; }
    public string CatidName { get; set; }
    public string Photo { get; set; }
}